druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -634,9 +634,19 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
         ImmutableList.Builder<JsonCollation> colBuilder =
             ImmutableList.builder();
         for (Pair<Integer, Direction> p : Pair.zip(collationIndexes, collationDirections)) {
+          boolean isNumericSort = false;
```

Comment:
Why don't we extract the _types_ in the loop in L464 into a list  _collationTypes_, we pass the list to _getQuery_ method, and then just use the following check?

``if (collationTypes.get(x).getType().getFamily() == SqlTypeFamily.NUMERIC) { ... } ``

That would be less ad-hoc than just using group by column position and then falling back to checking projects (plus it should work even if dimensions are not of character type in future).